azext_edge/edge/providers/check/mq.py

```diff
@@ -547,6 +674,328 @@ def evaluate_brokers(
+                authz_policies_eval_status = CheckTaskStatus.error.value
+
+            if (
+                detail_level != ResourceOutputDetailLevel.summary.value
+                or authz_policies_eval_status != CheckTaskStatus.success.value
+            ):
+                check_manager.add_display(
+                    target_name=target_authorizations,
+                    namespace=namespace,
+                    display=Padding(authz_policies_desc, (0, 0, 0, 12)),
+                )
+
+            check_manager.add_target_eval(
+                target_name=target_authorizations,
+                namespace=namespace,
+                status=authz_policies_eval_status,
+                value={"spec.authorizationPolicies": authz_policies},
+                resource_name=authz_name,
+            )
+
+            if authz_policies and detail_level == ResourceOutputDetailLevel.verbose.value:
+                process_dict_resource(
+                    check_manager=check_manager,
+                    target_name=target_authorizations,
+                    resource=authz_policies,
+                    namespace=namespace,
+                    padding=14,
+                )
+
+            check_manager.add_target_conditions(
+                target_name=target_authorizations,
+                namespace=namespace,
+                conditions=authz_conditions,
+            )
+
+    return check_manager.as_dict(as_list)
+
+
+def _evaluate_broker_reference(
+    check_manager: CheckManager,
+    owner_reference: dict,
+    target_name: str,
+    namespace: str,
+    resource_name: str,
+    added_condition: bool,
+) -> str:
+    broker_reference = [ref for ref in owner_reference if ref.get("kind").lower() == MqResourceKinds.BROKER.value]
+    if not broker_reference:
+        # skip this check
+        return ""
+
+    # should only have one broker reference
+    broker_reference_name = broker_reference[0].get("name")
+
+    if not added_condition:
+        check_manager.add_target_conditions(
+            target_name=target_name,
+            namespace=namespace,
+            conditions=["valid(brokerRef)"],
+        )
+        added_condition = True
+
+    valid_broker_refs = _get_valid_references(kind=MqResourceKinds.BROKER, namespace=namespace)
+    ref_eval_status = CheckTaskStatus.success.value
+    ref_eval_value = {}
+
+    if broker_reference_name not in valid_broker_refs:
+        ref_display = f"{colorize_string(color='red', value='Invalid')} broker reference {{{colorize_string(color='red', value=broker_reference_name)}}}."
+        ref_eval_status = CheckTaskStatus.error.value
+        ref_eval_value["valid(spec.brokerRef)"] = False
+    else:
+        ref_display = f"{colorize_string(color='green', value='Valid')} broker reference {{{colorize_string(color='green', value=broker_reference_name)}}}."
+        ref_eval_value["valid(spec.brokerRef)"] = True
```

Comment:
Similar nits as mentioned above re: breaking out the strings and values, but another consideration is there are some places we use this format:

`[resource_kind] reference {resource_name} is [valid/invalid]` - with the resource name standard cyan and only the valid/invalid colored as success / error.

And it looks like here it's:

`[Valid/Invalid] [resource_kind] reference {resource_name}` with both the valid/invalid and the resource_name colorized as green/red.

I'm not sure how many of these we have across, say, brokers and dataflows, but we should align them to be consistent